tracer/build/_build/Build.Steps.cs

```diff
@@ -613,23 +613,49 @@ async Task DownloadWafVersion(string libddwafVersion = null, string uncompressFo
         .Unlisted()
         .After(CompileManagedSrc)
         .Executes(() =>
-        {
-            var targetFrameworks = IsWin
-                ? TargetFrameworks
-                : TargetFrameworks.Where(framework => !framework.ToString().StartsWith("net4"));
-
-            // Publish Datadog.Trace.MSBuild which includes Datadog.Trace
-            DotNetPublish(s => s
-                .SetProject(Solution.GetProject(Projects.DatadogTraceMsBuild))
-                .SetConfiguration(BuildConfiguration)
-                .SetTargetPlatformAnyCPU()
-                .EnableNoBuild()
-                .EnableNoRestore()
-                .CombineWith(targetFrameworks, (p, framework) => p
-                    .SetFramework(framework)
-                    .SetOutput(MonitoringHomeDirectory / framework)));
-        });
-
+                                        {
+                                            if (PublishReadyToRun)
+                                            {
+                                                var targetFramework = TargetFramework.NET6_0;
+                                                Logger.Information($"ReadyToRun is {PublishReadyToRun}");
+
+                                                // Needed as we need to restore with the RuntimeIdentifier
+                                                DotNetRestore(s => s
+                                                                  .SetProjectFile(Solution.GetProject(Projects.DatadogTraceMsBuild))
+                                                                  .SetPublishReadyToRun(PublishReadyToRun)
+                                                                  .SetRuntime(RuntimeIdentifier)
+                                                );
+
+                                                DotNetPublish(s => s
+                                                                  .SetProject(Solution.GetProject(Projects.DatadogTraceMsBuild))
+                                                                  .SetConfiguration(BuildConfiguration)
+                                                                  .SetTargetPlatformAnyCPU()
+                                                                  .SetPublishReadyToRun(PublishReadyToRun)
+                                                                  .SetRuntime(RuntimeIdentifier)
+                                                                  .SetSelfContained(false)
+                                                                  .SetFramework(targetFramework)
+                                                                  .SetOutput(MonitoringHomeDirectory / targetFramework)
```

Comment:
`PublishReadToRun` makes it look like it's a "generic" switch, but then it does a very specific scenario for AWS (i.e. .NET 6.0 only). IMO, if we're going to just add a "publish r2r" option, then we should do it for _all_ TFMs. You're going to strip out the ones you don't need anyway, right?

Reply:
You're right, created it's own Target